packages/data/lib/src/repositories/auth_repository.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:domain/domain.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_facebook_auth/flutter_facebook_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+const _usersCollection = 'StarMovie_users';
+
+class AuthRepositoryImpl implements AuthRepository {
+  final FirebaseAuth firebaseAuth;
+  final FirebaseFirestore firebaseFirestore;
+  final FacebookAuth facebookAuth;
+
+  const AuthRepositoryImpl({
+    required this.firebaseAuth,
+    required this.firebaseFirestore,
+    required this.facebookAuth,
+  });
+
+  @override
+  Future<List<UserEmailPass>> fetchUsers() async {
+    final users = await firebaseFirestore.collection(_usersCollection).get();
+    return users.docs.map((e) => UserEmailPass.fromJson(e.data())).toList();
```

Comment:
we definitely don't need this method. what if we have 100 million users? instead of fetching whole table we should execute the query, like instead of

select * from users;

we should use

select * from users where email = 'email' AND password = 'password'